src/mailstoreobserver.cpp

```diff
@@ -229,29 +237,15 @@ QSharedPointer<MessageInfo> MailStoreObserver::constructMessageInfo(const QMailM
 
 // Check if this message should be notified, old messages are not
 // notified since QMailMessage::NoNotification is used
-bool MailStoreObserver::notifyMessage(const QMailMessageMetaData &message)
+bool MailStoreObserver::shouldNotify(const QMailMessageMetaData &message)
 {
-    if (message.messageType() == QMailMessage::Email &&
-        !(message.status() & QMailMessage::Read) &&
-        !(message.status() & QMailMessage::Temporary) &&
-        !(message.status() & QMailMessage::NoNotification) &&
-        !(message.status() & QMailMessage::Junk) &&
-        !(message.status() & QMailMessage::Trash) &&
-        messageInFolderToSync(message)) {
-        return true;
-    } else {
-        return false;
-    }
-}
-
-// App is on screen beep only
-void MailStoreObserver::notifyOnly()
-{
-    Notification notification;
-    initNotification(&notification);
-    notification.setIsTransient(true);
-    notification.setHintValue("x-nemo-feedback", QStringLiteral("email"));
-    notification.publish();
+    return message.messageType() == QMailMessage::Email
+            && !(message.status() & QMailMessage::Read)
+            && !(message.status() & QMailMessage::Temporary)
+            && !(message.status() & QMailMessage::NoNotification)
+            && !(message.status() & QMailMessage::Junk)
+            && !(message.status() & QMailMessage::Trash)
```

Comment:
nit: does not really matter, but is there some reason why each staus bit is tested separately? vs something like

    someType excludeMask = QMailMessage::Read | QMailMessage::Temporary | ...;
    ...
        && !(message.status() & excludeMask)
    ...

Reply:
Don't think is reason. Simplified.